### DISCUSSION TITLE
Running docker compose build throws Uncaught Exception: Invalid environment provided to "App\Kernel": the environment cannot be empty.

### DISCUSSION THREAD

**Discussion by rahuls-gfg** [symfony/symfony#49858] [Q&A]
Hello,

I have a project built using PHP, Symfony and Composer. I am completely new to entire PHP ecosystem. As of now, I am just trying to run the docker services  in my local pc using docker compose build. The command shows the the following error:

```
#0 7.838 Run composer recipes at any time to see the status of your Symfony recipes.
#0 7.838
#0 7.934 Executing script cache:clear [KO]
#0 7.980  [KO]
#0 7.980 Script cache:clear returned with error code 255
#0 7.980 !!  InvalidArgumentException {#57
#0 7.980 !!    #message: "Invalid environment provided to "App\Kernel": the environment cannot be empty."
#0 7.980 !!    #code: 0
#0 7.980 !!    #file: "./vendor/symfony/http-kernel/Kernel.php"
#0 7.980 !!    #line: 94
#0 7.980 !!    trace: {
#0 7.980 !!      ./vendor/symfony/http-kernel/Kernel.php:94 { …}
#0 7.980 !!      ./bin/console:14 {
#0 7.980 !!        {closure}
#0 7.980 !!        › return function (array $context) {
#0 7.980 !!        ›     $kernel = new Kernel($context['APP_ENV'], (bool) $context['APP_DEBUG']);
#0 7.980 !!        ›
#0 7.980 !!        arguments: {
#0 7.980 !!          $environment: ""
#0 7.980 !!          $debug: true
#0 7.980 !!        }
#0 7.980 !!      }
#0 7.980 !!      ./vendor/symfony/runtime/Resolver/DebugClosureResolver.php:28 { …}
#0 7.980 !!      ./vendor/autoload_runtime.php:30 { …}
#0 7.980 !!      ./bin/console:11 { …}
#0 7.980 !!    }
#0 7.980 !!  }
#0 7.980 !!  2023-03-29T11:46:06+00:00 [critical] Uncaught Exception: Invalid environment provided to "App\Kernel": the environment cannot be empty.
#0 7.980 !!
#0 7.980 Script @auto-scripts was called via post-install-cmd
```

Can anyone please help me on this issue? 

I have also attached file with detailed logs. [docker-compose-build.log](https://github.com/symfony/symfony/files/11100628/docker-compose.log)

Thank you.

**Comment by v-radev** [comment]
Can you add your .env file and your docker-compose.yaml?

**Reply by rahuls-gfg** [reply]
Here you go 
[docker-compose.txt](https://github.com/symfony/symfony/files/11110298/docker-compose.txt)
[env.txt](https://github.com/symfony/symfony/files/11110304/env.txt)

**Reply by v-radev** [reply]
Looks good to me. This error basically says that your app is not receiving the `APP_ENV` value, so you might want to check if the .env file is ok inside the container or if you are overwriting the environment variables somehow. Also check if running `bin/console cache:clear` inside the container gives you the same error or something else.

**Reply by rahuls-gfg** [reply]
Thanks for your reply. I will it and LYK.
